canonical-websites/snapcraft.io
# Cache-busting URLs for static files

I've done a few things here, but the main one is that I made it so static files will have URLs of the form:

`/static/file.ext?v=a23ef45`

The hash at the end will be generated from the contents of the file, so whenever the file's contents changes, the hash will change. This should prevent old (cached) CSS or JS being applied to new versions of the site. It should also mean that we can in future extend the cache time on static files (currently set to 12 hours - [the flask default](http://flask.pocoo.org/docs/0.12/api/#flask.Flask.default_config)).

Things done:

- [Fix ApiResponseErrorList error handling](https://github.com/nottrobin/snapcraft.io/commit/6061093f3776e8cb027a94654731c4aeec2ef0ad) - this was causing 500 error rather than 502 as it should
- [Use standard _layout.html template for homepage as well](https://github.com/nottrobin/snapcraft.io/commit/9d25773fcb6be9bc3e57344bd84cf6845053299d) - this implicitly removes the Google Optimize tracking snippet - @evandandrea let me know if this is a problem
- Fixes #438 - Badly cached CSS (and possibly JS)

QA
--

`./run`

Go to http://127.0.0.1:8004 and browse around a lot.

- Check CSS and JS are loading correctly
- Check homepage still looks the same
